py/gpu_specter/extract/both.py

```diff
@@ -0,0 +1,149 @@
+import numpy as np
+
+try:
+    import cupy as cp
+    cupy_available = True
+except ImportError:
+    cupy_available = False
```

Comment:
This module is for functions that work with both cupy and numpy without knowing or caring which type of array they are working on, right?  If that is correct, please add a top level comment explaining the purpose of this module vs. `extract.cpu` and `extract.gpu`.

Also, this `cupy_available` check appears to be a leftover that isn't needed in this particular module.

Reply:
Yes, that's correct. I added a brief comment at the top of `extract.both`. I'm still thinking through the overall organization. I think the process of enabling the gpu option in spex will be helpful in determining the appropriate boundaries.